src/pybind/mgr/call_home_agent/module.py

```diff
@@ -0,0 +1,322 @@
+
+
+def report_inventory() -> str:
+    """
+    Produce the inventory report
+    """
+    dt = datetime.timestamp(datetime.now())
+    # Set timestamps
+    reports_header['event_time'] = datetime.fromtimestamp(dt).strftime("%Y-%m-%d %H:%M:%S")
+    reports_header['event_time_ms'] = str(int(dt))
+    reports_header['local_event_time'] = datetime.fromtimestamp(dt).strftime("%a %b %d %H:%M:%S %Z")
+
+    # Set event id
+    reports_header['event_id'] = "IBM_chc_event_RedHatMarine_ceph_{}_daily_report_{}".format(
+                                            reports_header['asset_virtual_id'],
+                                            reports_header['event_time_ms'])
+
+    # return json.dumps({**reports_header, **inventory()}) <---- TODO ADD INVENTORY (from datacollector) as event
+    return json.dumps(reports_header)
+
+
+def report_performance() -> str:
+    """
+    TODO: temporally uses inventory report
+    """
+    return report_inventory()
+
+
+class Report:
+    def __init__(self, name: str, fn: Callable[[], str], url: str, minutes_interval: int):
+        self.name = name                       # name of the report
+        self.fn = fn                           # function used to retrieve the data
+        self.url = url                         # url to send the report
+        self.interval = minutes_interval * 60  # interval to send the report (seconds)
+
+    def print(self) -> str:
+        if reports_header['private_key'] and reports_header['api_key']:
+            return self.fn()
+        else:
+            raise Exception('Not able to print <%s> report. Identification keys are not available' % self.name)
+
+    def send(self) -> str:
+        fail_reason = ""
+        resp = None
+        if reports_header['private_key'] and reports_header['api_key']:
+            try:
+                resp = requests.post(url=self.url,
+                                     headers={'accept': 'application/json', 'content-type': 'application/json'},
+                                     data=self.fn())
+                resp.raise_for_status()
+            except Exception as e:
+                explanation = resp.content if resp else ''
+                fail_reason = 'Failed to send <%s> to <%s>: %s \n%s' % (self.name, self.url, str(e), explanation)
+        else:
+            fail_reason = 'Not able to send <%s> report. Identification keys are not available' % self.name
+        return fail_reason
+
+
+class CallHomeAgent(MgrModule):
+    MODULE_OPTIONS: List[Option] = [
+        Option(
+            name='ches_endpoint',
+            type='str',
+            default=CHES_ENDPOINT,
+            desc='Call Home Event streamer end point'
+        ),
+        Option(
+            name='interval_inventory_report_minutes',
+            type='int',
+            default=INTERVAL_INVENTORY_REPORT_MINUTES,
+            desc='Time frequency for the inventory report'
```

Comment:
This is what users see in the Dashboard's Mgr-Module section, so I'd use the most intuitive explanation:
```suggestion
            desc='Minutes to wait before submitting a new report'
```